entity-management-web/src/main/java/eu/europeana/entitymanagement/web/service/impl/EntityRecordService.java

```diff
@@ -106,6 +113,133 @@ public EntityRecord createEntityFromRequest(EntityPreview entityCreationRequest,
 	return Optional.empty();
     }
 
+    public void performGlobalReferentialIntegrity(EntityRecord entityRecord) throws JsonMappingException, JsonProcessingException {
```

Comment:
the complexity of this method is too high, please refactor the code by extracting the code from the case statements in individual methods

Reply:
Done.